### PR TITLE
feat(money-field): add high-precision price badge

### PR DESCRIPTION
**Pull request by dferber90**
- Removes the ability to pass in a badge manually
- Adds the `showHighPrecisionBadge` prop which will show the "high precision price" badge in case a high-precision price was entered.

hppbadge

### CODE REVIEW

File: src/components/fields/money-field/money-field.form.story.js [hunk 1]
```diff
@@ -66,6 +73,20 @@ storiesOf('Examples|Forms/Fields', module)
                     }
                   }}
                 />
+                <MoneyField
+                  title="Price per ton (High Precision)"
+                  description={'How much is a ton of fish?'}
```
Comment (tdeekens) [inline src/components/fields/money-field/money-field.form.story.js]: I like this.

File: src/components/fields/money-field/money-field.js [hunk 2]
```diff
@@ -111,7 +122,20 @@ class MoneyField extends React.Component {
             description={this.props.description}
             onInfoButtonClick={this.props.onInfoButtonClick}
             hintIcon={this.props.hintIcon}
-            badge={this.props.badge}
+            badge={
+              this.props.showHighPrecisionBadge &&
```
Comment (tdeekens) [inline src/components/fields/money-field/money-field.js]: `shouldShowHighPrecisionBadge`. I know it gets long. Maybe these prefixes for React props are something we should really drop.

**Comment by lufego** [comment]
Code looks good. I have a user related question to design team tho. What if the user doesn't know what a "high precision price" is about? maybe we can add an info button or a tooltip on hover explaining it?

**Comment by dferber90** [comment]
> Code looks good. I have a user related question to design team tho. What if the user doesn't know what a "high precision price" is about? maybe we can add an info button or a tooltip on hover explaining it?

Nice catch. I spoke with Filip. The text in the designs is coupled tightly to the views. We will keep it as a pure indicator for now. We will open a modal eventually which will explain the meaning of a high-precision price for the specific content. To support this, the `MoneyField` will get an `onHighPrecisionBadgeClick`. Then the consumers can show the modal with any text they like.

File: src/components/fields/money-field/money-field.js (hunk 2 above)
Comment (dferber90) [reply src/components/fields/money-field/money-field.js]: Should I do it or not? I thought the `show`  prefix is as a good enough indicator. But it does sound like a function now that you mention it.

Comment (dferber90) [reply src/components/fields/money-field/money-field.js]: Maybe just `hasHighPrecisionBadge`. The potentially confusing part is that the badge will only be shown when the value holds a high precision price. But the `show` prefix has the same problem.

**Comment by dferber90** [comment]
Handed off to @filippobocik for UX review (can't assign as he's not part of the repo yet).

**Comment by dferber90** [comment]
I'll merge now. We'll address UX feedback in a new PR. I want to do a refactoring now which would be inconvenient without having this component in.

**Pull request merged**